File: coredata/_HealthDataWrapper.m
#import "_HealthDataWrapper.h"
#import "HealthData.h"

@interface _HealthDataWrapper ()
@end

@implementation _HealthDataWrapper {
}
- (instancetype)initWithEntity:(HealthData *)entity {
    self = [super init];
    if (self) {
        _entity = entity;
        self.height = entity.height;
        self.timeStamp = entity.timeStamp;
        self.weight = entity.weight;
    }
    return self;
}

- (void)updateEntity:(HealthData *)entity {
    entity.height = self.height;
    entity.timeStamp = self.timeStamp;
    entity.weight = self.weight;
}
@end

File: coredata/Swift/_HealthDataWrapper.swift
import Foundation

class _HealthDataWrapper: NSObject {
    let _entity: HealthData?
    func entity() -> HealthData? {
        return _entity
    }
    var height: NSNumber?
    var timeStamp: NSDate?
    var weight: NSNumber?

    init(healthData: HealthData?) {
        _entity = healthData
        super.init()
        if let e = healthData {
            self.height = e.height
            self.timeStamp = e.timeStamp
            self.weight = e.weight
        }
    }
    func updateHealthData(healthData: HealthData) {
        healthData.height = self.height
        healthData.timeStamp = self.timeStamp
        healthData.weight = self.weight
    }
}
